sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ColumnVector.java

```diff
@@ -285,19 +285,19 @@ public void reserve(int requiredCapacity) {
         try {
           reserveInternal(newCapacity);
         } catch (OutOfMemoryError outOfMemoryError) {
-          throwUnsupportedException(newCapacity, requiredCapacity, outOfMemoryError);
+          throwUnsupportedException(requiredCapacity, outOfMemoryError);
         }
       } else {
-        throwUnsupportedException(newCapacity, requiredCapacity, null);
+        throwUnsupportedException(requiredCapacity, null);
       }
     }
   }
 
-  private void throwUnsupportedException(int newCapacity, int requiredCapacity, Throwable cause) {
-    String message = "Cannot reserve more than " + newCapacity +
-        " bytes in the vectorized reader (requested = " + requiredCapacity + " bytes). As a" +
-        " workaround, you can disable the vectorized reader by setting "
-        + SQLConf.PARQUET_VECTORIZED_READER_ENABLED().key() + " to false.";
+  private void throwUnsupportedException(int requiredCapacity, Throwable cause) {
+    String message = "Cannot reserve additional contiguous bytes in the vectorized reader " +
+        "(requested = " + requiredCapacity + " bytes). As a workaround, you can disable the " +
+        "vectorized reader by setting " + SQLConf.PARQUET_VECTORIZED_READER_ENABLED().key() +
+        " to false.";
```

Comment:
oh, I was thinking if we can explain the reason that fails the allocation instead of just saying we cannot allocate mem.